src/types/user.types.ts

```diff
@@ -43,3 +43,10 @@ export type RelatedUsers = {
   numberOfElements: number;
   empty: boolean;
 };
+
+export type UserBlockInfo = {
+  isBlocked: boolean;
+  whichUserBlocked:
+    | { name: string; surname: string; email: string }
+    | undefined;
```

Comment:
Jak dodasz to jako opcjonalny atrybut za pomocą `?` to wtedy z automatu będzie `| undefined`
```suggestion
  whichUserBlocked?: { 
    name: string;
    surname: string;
    email: string;
  };
```